config/src/main/java/com/typesafe/config/impl/ConfigConditional.java

```diff
@@ -0,0 +1,36 @@
+package com.typesafe.config.impl;
+
+import com.typesafe.config.ConfigException;
+
+import java.util.Map;
+
+public class ConfigConditional {
+
+    private SubstitutionExpression left;
+    private AbstractConfigValue right;
+    private SimpleConfigObject body;
+
+    ConfigConditional(SubstitutionExpression left, AbstractConfigValue right, SimpleConfigObject body) {
+        this.left = left;
+        this.right = right;
+        this.body = body;
+
+        if (this.left.optional()) {
+            throw new ConfigException.BugOrBroken("Substitutions in conditional expressions cannot be optional");
```

Comment:
Could we attempt more contextual error messages? Like can you add the value to of this.left to the message? Not really sure what optional means in this context either